src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -220,6 +230,28 @@ func (s *fileSystemSource) Read(
 	return results, nil
 }
 
+func (s *fileSystemSource) runMigrations(infoFilesByNamespace map[namespace.Metadata]fs.InfoFileResultsPerShard) {
```

Comment:
nit: worth adding a specific span on here? Also think it may be worth logging how long migration takes here

Reply:
Yes, to all of this :). Will investigate!